### ISSUE TITLE
Progress report

### ISSUE THREAD

**Issue by bosr** [ott-jax/ott#210]
Hi, for computations than can take several hours, it would be great if OTT-jax reported progress. I can see at least two types of progress reports:

1. In some computations, the number of steps is known in advance, for instance if using the optional `batch_size` parameter of `PointCloud` and using `Sinkhorn` or `LRSinkhorn`: each Sinkhorn iteration is a fixed set of submatrices operations.

2. In some other cases the number of steps is indeterminate because we're waiting for some convergence criterion. But OTT-jax could in these cases report the number of iterations performed so far and the value associated with the convergence criterion, same as you would report a loss in a training loop.

Because reporting progress is a side effect, jax has introduced the [`host_callback`](https://jax.readthedocs.io/en/latest/jax.experimental.host_callback.html) module. I think it's worth a try.

I don't think OTT-jax should depend on progress report libs; it would be instead the responsibility of the user to provide their own callback function, such as tqdm.

Would you be open to this?

**Comment by marcocuturi** [comment]
I think that would be definitely very useful for people that have a vague idea of how long they should wait in front of their sinkhorn haha. i think it's very cool.

**Comment by michalk8** [comment]
closed via #228
